plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,117 @@
+package nodeenv
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeEnvironment", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewPodNodeEnvironment(client, nil)
+	})
+}
+
+const (
+	NamespaceNodeSelector = "kubernetes.io/node-selector"
+	DefaultNodeSelector   = ""
```

Comment:
This should be something the cluster admin can set when configuring the cluster. It needs to come from `config` on line 15 above